src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/CSharpVirtualDocumentFactory.cs

```diff
@@ -15,94 +14,43 @@ namespace Microsoft.VisualStudio.LanguageServerClient.Razor
 {
     [Shared]
     [Export(typeof(VirtualDocumentFactory))]
-    internal class CSharpVirtualDocumentFactory : VirtualDocumentFactory
+    internal class CSharpVirtualDocumentFactory : VirtualDocumentFactoryBase
     {
-        private readonly IContentTypeRegistryService _contentTypeRegistry;
-        private readonly ITextBufferFactoryService _textBufferFactory;
-        private readonly ITextDocumentFactoryService _textDocumentFactory;
-        private readonly FileUriProvider _fileUriProvider;
-        private IContentType _csharpLSPContentType;
+        private static readonly Dictionary<object, object> _additionalLanguageBufferProperties = new Dictionary<object, object>
```

Comment:
```suggestion
        private static readonly IReadOnlyDictionary<object, object> _languageBufferProperties = new Dictionary<object, object>
```

Reply:
It's private, but sure, why not. 

---
In reply to: [452550213](https://github.com/dotnet/aspnetcore-tooling/pull/2201#discussion_r452550213) [](ancestors = 452550213)